### PR TITLE
feat: New contextual modal

### PR DESCRIPTION
**Pull request by kbesingeryeh**
# [SC-22988](https://app.shortcut.com/homebound-team/story/22988/new-contextual-modal-beam-component)

* Added a new `ContextualModal` that an engineer/designer can use on the front-end to render whatever content they'd like after clicking on a Button. The modal allows us to reuse the same container, but put different things inside.

Screen Shot 2022-10-20 at 1 37 18 PM

### CODE REVIEW

File: src/components/ButtonMenu.tsx
Comment (bdow) [inline src/components/ButtonMenu.tsx:89]: **suggestion:** Would prefer this to be its own file. Maybe in `src/components/internal/ContextualModal.tsx`.

File: src/components/ButtonMenu.tsx
Comment (bdow) [inline src/components/ButtonMenu.tsx:95]: **suggestion:** Could we use the `useTestIds` hook here? Something like:
```
const tid = useTestIds(props, "popup"); 

<div {...tid.title}>
```

File: src/components/ButtonMenu.tsx
Comment (bdow) [inline src/components/ButtonMenu.tsx:99]: Same thing regarding test ids here.

File: src/components/internal/OverlayTrigger.tsx
Comment (bdow) [inline src/components/internal/OverlayTrigger.tsx:118]: **suggestion**: Can we move this util to the ButtonMenu component?

File: src/components/internal/OverlayTrigger.tsx
Comment (bdow) [inline src/components/internal/OverlayTrigger.tsx:116]: **suggestion:** When we do these typeguards we should be specify what `prop` could be. Something like:
```
function isItemsProps(props: ButtonMenuProps | SomeOtherProps): props is ButtonMenuProps {
```

Comment (kbesingeryeh) [reply src/components/internal/OverlayTrigger.tsx:116]: got it because its not only using `WithMenuItemsProps` but also uses `ButtonMenuBaseProps`

File: src/components/ButtonMenu.tsx
Comment (kbesingeryeh) [reply src/components/ButtonMenu.tsx:89]: got it. I noticed that when I created a file called `ContextualModal` Webstorm changes the file to a `text` file instead of `tsx`... if I rename the file to `ConTextualModal` it keeps the file as `tsx` but its kind of strange naming convention...

File: src/components/ButtonMenu.stories.tsx
Comment (stephenh) [reply src/components/ButtonMenu.stories.tsx:96]: cc @bdow for naming thoughts

File: src/components/ButtonModal.tsx
Comment (stephenh) [inline src/components/ButtonModal.tsx:13]: Suggestion: Probably rename this to `ButtonModalProps` so that it matches the component name?

File: src/components/ButtonModal.tsx
Comment (stephenh) [inline src/components/ButtonModal.tsx:18]: Suggestion: Maybe prefix this like `storybookDefaultOpen` so that engineers using the component, who may not see the in-source comment, easily know "ah yeah I don't need to worry about this".

File: src/components/ButtonModal.tsx
Comment (bdow) [inline src/components/ButtonModal.tsx:13]: **nit:** Can you call this `ButtonModalProps`?

**Review by stephenh — COMMENTED** [review]
Will defer to @bdow for the +1, few suggestions, but otherwise lgtm!

File: src/components/internal/ContextualModal.tsx
Comment (bdow) [inline src/components/internal/ContextualModal.tsx:11]: **question:** What is the `myPx(4)` doing? Also curious about the `maxh("inherit")`.

Comment (kbesingeryeh) [reply src/components/internal/ContextualModal.tsx:11]: I was looking at the `Menu` component and trying to match certain properties, but I think I can remove the `margin-top/margin-bottom`... 

I added the `maxh("inherit")` from a comment in the `Menu` component that mentioned using it to allow us to take advantage of the height set on the overlay container, so I thought it made sense to use it here.

File: src/components/ButtonModal.tsx
Comment (kbesingeryeh) [reply src/components/ButtonModal.tsx:18]: got it. ty!

File: src/components/ButtonModal.tsx
Comment (kbesingeryeh) [reply src/components/ButtonModal.tsx:13]: yes, will do.

**Comment by kbesingeryeh** [comment]
@aeppy @kgrant211 @jeffreyqua here is the storybook link: https://60466f7d43c37c0021788867-brlzghvjsy.chromatic.com/?path=/story/workspace-components-button-modal--button-modal-with-title

let me know if it doesn't work!

File: src/components/internal/ContextualModal.tsx
Comment (bdow) [reply src/components/internal/ContextualModal.tsx]: Sounds good.

**Review by bdow — COMMENTED** [review]
Before to merging, please have @jeffreyqua QA the changes in the storybook built by this PR. Once he gives the 👌 , then we can merge it in.

File: src/components/internal/ContextualModal.tsx
Comment (kbesingeryeh) [reply src/components/internal/ContextualModal.tsx:20]: @jeffreyqua can you confirm if this is needed? I will add it in if so, also please let me know if this is approved. this modal is blocking 4 other stories!

Comment (jeffreyqua) [reply src/components/internal/ContextualModal.tsx:20]: @kbesingeryeh We can remove that. I don't think that's needed. I will update the mocks.

**Comment by jeffreyqua** [comment]
This looks good to me. Let's 🍣 !

**Review by kgrant211 — APPROVED** [review]
Looks good to me! Thanks @kbesingeryeh

**Comment by aeppy** [comment]
Thanks @kbesingeryeh good to me!

**Pull request merged**
